### PR TITLE
ci: exprimental lualint integration

### PR DESCRIPTION
**Pull request by pluveto**

### CODE REVIEW

**Comment by pluveto** [comment]
Only if secret token is added to the setting, this pull request will pass the check.

File: .github/workflows/lualint.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+
+      - name: Download Lualint
```
Comment (wheelerlaw) [inline .github/workflows/lualint.yml:24]:
```suggestion
      - name: Generate token
        id: github-app-token
        uses: tibdex/github-app-token@021a2405c7f990db57f5eae5397423dcc554159c # v1.7.0
        with:
          app_id: ${{ vars.KONG_GH_APP_ID }}
          private_key: ${{ secrets.KONG_GH_APP_PRIVATE_KEY }}

      - name: Download Lualint
```

Comment (wheelerlaw) [inline .github/workflows/lualint.yml:24]:
```suggestion
      - name: Generate token
        id: github-app-token
        uses: tibdex/github-app-token@021a2405c7f990db57f5eae5397423dcc554159c # v1.7.0
        with:
          app_id: ${{ vars.KONG_GH_APP_ID }}
          private_key: ${{ secrets.KONG_GH_APP_PRIVATE_KEY }}

      - name: Download Lualint
```

File: .github/workflows/lualint.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        uses: robinraju/release-downloader@v1.7
+        with:
+          repository: "kong/lualint"
+          token: ${{ secrets.LUALINT_REPO_ACCESS_TOKEN }}
```
Comment (wheelerlaw) [inline .github/workflows/lualint.yml:28]:
```suggestion
          token: ${{ steps.github-app-token.outputs.token }}
```

**Review by wheelerlaw — COMMENTED** [review]
I've installed the Kong GitHub app to this repo and the lualint repo, and granted it read/write access to repository contents (including downloads from releases). 

This is how you use the app to generate an ephemeral token that can be used in place of a PAT.

File: .github/workflows/lualint.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        uses: robinraju/release-downloader@v1.7
+        with:
+          repository: "kong/lualint"
+          token: ${{ steps.github-app-token.outputs.token }}
+          latest: true
+          fileName: "*_x86_64-unknown-linux-musl.tar.gz"
+
+      - name: Lint all changed lua files
+        run: |
+          tar -zvxf lualint_x86_64-unknown-linux-gnu.zip;
```
Comment (wheelerlaw) [inline .github/workflows/lualint.yml:41]:
```suggestion
          tar -zvxf lualint_v0.1.0-1_x86_64-unknown-linux-musl.tar.gz;
```

**Review by wheelerlaw — COMMENTED** [review]
Got the token generation [working](https://github.com/Kong/atc-router/actions/runs/4321152101/jobs/7555183776#step:5:13), the workflow just needs a small tweak in order to use the downloaded artifact.

**Comment by pluveto** [comment]
> Got the token generation [working](https://github.com/Kong/atc-router/actions/runs/4321152101/jobs/7555183776#step:5:13), the workflow just needs a small tweak in order to use the downloaded artifact.

Brilliant! It works quite well.

File: .github/workflows/lualint.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+name: CI
```
Comment (dndx) [inline .github/workflows/lualint.yml:1]:
```suggestion
name: Lua lint```

File: .github/workflows/lualint.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Generate token
+        id: github-app-token
+        uses: tibdex/github-app-token@021a2405c7f990db57f5eae5397423dcc554159c # v1.7.0
+        with:
+          app_id: ${{ vars.KONG_GH_APP_ID }}
+          private_key: ${{ secrets.KONG_GH_APP_PRIVATE_KEY }}
+
+      - name: Download Lualint
+        uses: robinraju/release-downloader@v1.7
```
Comment (dndx) [inline .github/workflows/lualint.yml:32]: We need to pin by commit hash for security reasons. See: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

File: .github/workflows/lualint.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+          repository: "kong/lualint"
+          token: ${{ steps.github-app-token.outputs.token }}
+          latest: true
+          fileName: "lualint_x86_64-unknown-linux-gnu.zip"
+
+      - name: Lint all changed lua files
+        run: |
+          unzip lualint_x86_64-unknown-linux-gnu.zip;
+          chmod +x ./lualint;
+          for file in ${{steps.changed-files.outputs.all_changed_files}}; do
```
Comment (dndx) [inline .github/workflows/lualint.yml:43]:
```suggestion
          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
```

We should be consistent over styles of above.

File: .github/workflows/lualint.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+                  "one_line_before_else": {},
+                  "eof_blank_line": {},
+                  "table_ctor_comma": {},
+                  "func_separation": {}
+              }' "$file";
+              if [[ $? -ne 0 ]]; then
+                ALL_PASSED="false";
+              fi;
+            fi;
+          done;
```
Comment (dndx) [inline .github/workflows/lualint.yml:58]:
```suggestion
          done;

```

**Review by dndx — CHANGES_REQUESTED** [review]

File: .github/workflows/lualint.yml (hunk 5 above)
Comment (pluveto) [reply .github/workflows/lualint.yml]: done

File: .github/workflows/lualint.yml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+          app_id: ${{ vars.KONG_GH_APP_ID }}
+          private_key: ${{ secrets.KONG_GH_APP_PRIVATE_KEY }}
+
+      - name: Download Lualint
+        uses: robinraju/release-downloader@768b85c8d69164800db5fc00337ab917daf3ce68
+        with:
+          repository: "kong/lualint"
+          token: ${{ steps.github-app-token.outputs.token }}
+          latest: true
+          fileName: "lualint_x86_64-unknown-linux-gnu.zip"
```
Comment (chronolaw) [inline .github/workflows/lualint.yml:39]: only x86_64 build? what will happen if ci runs on arm64?

**Comment by pluveto** [comment]
Please merge the action version. NOT THIS.

**Pull request closed without merging**